src/plopp/graphics/mesh.py

```diff
@@ -106,77 +99,33 @@ def __init__(self,
         elif vmax is not None:
             self._extend = 'max'
 
-        self._make_mesh(**kwargs)
-
-    def _make_mesh(self, shading='auto', rasterized=True, **kwargs):
         x, y, z = _from_data_array_to_pcolormesh(data=self._data.data,
                                                  coords=self._bin_edge_coords)
         self._mesh = self._ax.pcolormesh(x.values,
                                          y.values,
                                          z.values,
-                                         cmap=self.color_mapper.cmap,
                                          shading=shading,
                                          rasterized=rasterized,
                                          **kwargs)
-        if self._cbar:
-            self._cbar = colorbar(self._mesh,
-                                  ax=self._ax,
-                                  cax=self._cax,
-                                  extend=self._extend,
-                                  label=name_with_unit(var=self._data.data, name=""))
-
-            # Add event that toggles the norm of the colorbar when clicked on
-            # TODO: change this to a double-click event once this is supported in
-            # jupyterlab, see https://github.com/matplotlib/ipympl/pull/446
-            self._cbar.ax.set_picker(5)
-            self._ax.figure.canvas.mpl_connect('pick_event', self.toggle_norm)
-            self._cbar.ax.yaxis.set_label_coords(-1.1, 0.5)
         self._mesh.set_array(None)
-        self._set_norm()
-        self._set_mesh_colors()
 
-    def _set_clim(self):
-        self._mesh.set_clim(self.color_mapper.vmin, self.color_mapper.vmax)
-
-    def _set_mesh_colors(self):
-        to_mapper = sc.DataArray(data=_maybe_repeat_values(
-            data=self._data.data, coords=self._bin_edge_coords))
+    @property
+    def data(self):
+        out = sc.DataArray(data=_maybe_repeat_values(data=self._data.data,
+                                                     coords=self._bin_edge_coords))
         if self._data.masks:
-            to_mapper.masks['one_mask'] = _maybe_repeat_values(
-                data=sc.broadcast(merge_masks(self._data.masks),
-                                  dims=self._data.dims,
-                                  shape=self._data.shape),
-                coords=self._bin_edge_coords)
-
-        rgba = self.color_mapper.rgba(data=to_mapper)
+            out.masks['one_mask'] = _maybe_repeat_values(data=sc.broadcast(
+                merge_masks(self._data.masks),
+                dims=self._data.dims,
+                shape=self._data.shape),
+                                                         coords=self._bin_edge_coords)
+        return out
+
+    def set_colors(self, rgba):
         self._mesh.set_facecolors(rgba.reshape(np.prod(rgba.shape[:-1]), 4))
 
     def update(self, new_values: sc.DataArray):
         """
         Update image array with new values.
         """
         self._data = new_values
```

Comment:
Is update on mesh still needed?

Reply:
Yes, this needs to remain because we get the `.data` from the mesh to send to the `rgba`, and that `.data` could be different from the `new_values` in the case of a 2D coord.